pkg/api/artifact.go

```diff
@@ -32,6 +32,10 @@ type ArtifactDetails struct {
 	SHA256Sum string
 	// DownloadURL points to the target image.
 	DownloadURL string
+	// ImageArchitecture is the target architecture of the image.
+	ImageArchitecture string
+	// ImageOS is the target operating system of the image.
+	ImageOS string
```

Comment:
I don't think we need to have `ImageOS` in the Artifact. It is used for the registry image's OS field, which in our case will always be `linux`.